python/cudf/cudf/core/udf/groupby_utils.py

```diff
@@ -202,3 +203,30 @@ def jit_groupby_apply(offsets, grouped_values, function, *args):
     specialized[ngroups, tpb](*launch_args)
 
     return output
+
+
+def _jit_groupby_eligible(frame, func, args):
+    return (not frame.has_nulls) and _can_be_jitted(frame, func, args)
+
+
+def _can_be_jitted(frame, func, args):
+    """
+    Determine if this UDF is supported through the JIT engine
+    by attempting to compile just the function to PTX using the
+    target set of types
+    """
+    np_field_types = np.dtype(
+        list(
+            _supported_dtypes_from_frame(
+                frame, supported_types=SUPPORTED_GROUPBY_NUMPY_TYPES
+            ).items()
+        )
+    )
+    dataframe_group_type = _get_frame_groupby_type(
+        np_field_types, frame.index.dtype
+    )
+    try:
+        _get_udf_return_type(dataframe_group_type, func, args)
```

Comment:
Not necessary for this PR, but now that we're calling this function (`_get_udf_return_type`) more often is there a way to make it also cache the results like `_compile_udf` does? I suspect that this function isn't the cheapest.

Reply:
this goes through `cudautils.compile_udf` which also caches, though it's not the same cache :(. One of the caches holds the result of compiling the UDF separately from any kernel for the purposes of getting the return type and the other holds the compilation result of the full kernel.

Reply:
Yup I realize that they're two different functions with one calling the other. I'm wondering whether we should add a cache for this function as well in case there is significant overhead in the parts of this function _aside_ from the call to `compile_udf` since now we're calling this one (possibly a lot) more frequently.

Reply:
ah I see what you mean.

I think it'd certainly be worth going through where we're caching things and doing some profiling while we assess where it's the most useful to do so at this point, given the proliferation of calls to numba's compilation machinery in our udf code. Especially as we eventually get on disk caching too. I'll raise a separate issue about this.